### PR TITLE
API changes.

### PR DESCRIPTION
**Pull request by MilaDog**
Here are the changes:

1. Get all achievements `/achievement` -> `/achievements`
2. Company linked to a User account.
3. Cannot create a Company without an account.
4. Return Company details after creation.
5. Shop Pagination sort change from requiring a list to just a string.
6. User account ID is now a `str` and not an `int`
7. Fetching a Company gets the User's active company, if they have one.
8. Fetching a Company or User profile by user Discord ID. (any of these endpoints)
9. Endpoints documented in `README`

### CODE REVIEW

**Comment by iauu** [comment]
Include `id` and `created` in `GET /companies`

**Comment by MilaDog** [comment]
> Include `id` and `created` in `GET /companies`

Done

File: api/src/classes/company.py [hunk 1]
```diff
@@ -28,19 +29,15 @@ def __init__(self, session: Session, company: Company) -> None:
...
 
         :param company_id: ID of the Company to search for.
         :param session: Database session.
-        :param name: Name of the company.
-        :param owner_id: ID of the Owner.
         :return: Instance with the fetched company.
         """
         fetched_company: Company | None = session.exec(
-            select(Company).where(or_(Company.id == company_id, Company.name == name, Company.owner_id == owner_id))
+            select(Company).where(Company.owner_id == company_id, not_(Company.is_bankrupt))
```
Comment (iauu) [inline api/src/classes/company.py:40]: Possibly try to find the latest company if nothing match this query (i.e. all company the user own is bankrupt)

**Review by i-am-unknown-81514525 — APPROVED** [review]
Well, just a thought, could just be merged.

File: api/src/classes/company.py (hunk 1 above)
Comment (MilaDog) [reply api/src/classes/company.py:40]: Other functions rely on this one to get the user's active company for updates, deleting, purchasing etc.

I could possibly add an endpoint to return all the companies that a user has had? Like `/user/{id}/companies`?

Comment (iauu) [reply api/src/classes/company.py:40]: Maybe, that's just a small thought and adding an additional endpoint would be unlikely to be used.
The only reason I could think about is getting the company why after they go bankrupt to tell the user (but maybe that could be done just before caching)

So... Just ignore this :)

**Pull request merged**